include/sys/zstd/zstd.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License, Version 1.0 only
+ * (the "License").  You may not use this file except in compliance
+ * with the License.
+ *
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or http://www.opensolaris.org/os/licensing.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
```

Comment:
This makes no sense as the license header. The underlying implementations are not CDDL, this header should not be licensed in the same manner.

Reply:
Licence is on a per-file basis.

The underlaying implementation (module/zstd/zstd.c) is also CDDL and referenced here:
```
int zstd_init(void);
void zstd_fini(void);
```

We only view the code from the ZSTD project as being non CDDL.

Reply:
> I would love to have a discussion with you about copyright virality some day, but this is not the place and this is not a case where virality applies.

This has very little to do with that. And please don't use such pejorative terms. This has to do with the fact that code around zstd should remain under its original or similar terms. @allanjude did the right thing here with the rest of his code. I see no reason we can't respect the spirit here with the rest of it.

Reply:
> "And please don't use such pejorative terms"

Sorry but in copyright law virality is frequently used without any pejorative meaning

> "This has to do with the fact that code around zstd should remain under its original or similar terms."

You create an unworkable situation with this request, as we can turn it the other way around and say this should follow the licence on zstd.c (which is CDDL). Virality can work in two directions and we have made a clear choice which is neither wrong nor right. I'll leave it at that.

> "@allanjude did the right thing here with the rest of his code"

His code is also mostly CDDL too.

> "I see no reason we can't respect the spirit here with the rest of it."

This is no code feedback but your personal preference which fine is noted. I do not feel the need to discuss it more than I already did, unless the maintainers define it as formally wrong (which it legally isn't).

Reply:
Note: We also need all authors to agree with changing licences.
Previous tries to do so with ZSTD.c has failed

Reply:
> Sorry but in copyright law virality is frequently used without any pejorative meaning

The official term is "reciprocity". The word "virality" is pejorative. Anyway, meh.

Reply:
@c0d3z3r0 I think this can be marked as resolved...
Mixed: "wont fix"/"cant fix" is also a solution to a problem.

Reply:
I'm no lawyer (just really well-informed as needed to do FOSS work), but most treatises and other writings on it that use the term usually have a negative slant. 🤷‍♂ 

Anyway, whatever.

Reply:
> @allanjude did the right thing here with the rest of his code.

huh? Look at `module/zstd/zstd.c` in his PR, which is CDDL. Since `include/sys/zstd/zstd.h` is the header for `module/zstd/zstd.c`, it should be CDDL, too.

Reply:
@c0d3z3r0 do you mind that he might be confused since zstd. h of our zstd.c has the same name as the api header zstd.h of the library itself?

Reply:
Anyway, What ever it might be...
This is resolved, because even if he was right: 
- Its not required to change it
- It requires four(!) parties to consent

So this is resolved: Unless we recieve writen consent of all 4 authors, we cant retroactively change the licence on a file.

Reply:
from the codestyle and to avoid inclusion of wrong headers in c code it makes sense to rename it. i suggest zstd_impl.h

Reply:
@brainslayer I think that is more related to his issue with module include naming than the copyright header.... Please move this there, so we can close this discussion as resolved because there is no (solveable) copyright issue.

Reply:
@Ornias1993 yes i'm not talking about the copyright issue since there wasnt any. i just told that it should be renamed to avoid confusion. i can open a new review comment on the same line of course. but is this neccessary?

Reply:
You can, but it isn't needed. The file isn't called "zstd.h" the file is called 
`include/sys/zstd/zstd.h`

I looked into it and the name we use for the module internally is not zstd_impl but: zzstd.

I think it's pretty clear:
Everything NOT in /contrib/zstd is zfs side.
It's in the basic structure already...

Ask yourself:
Do we need to add "impl" to every middleware?
Or is it enough to not have it inside contrib?

Reply:
Simply put: renaming it isn't needed.
If a maintainer wants it renamed, fine. But lets follow the current convention and keep it like this. As this is in-line with how other modules are implemented.

Reply:
Anyway @c0d3z3r0 Mind flagging this as resolved?